### PR TITLE
Set up backend content manager 

### PR DESCRIPTION
**Pull request by DenisaCG**
This PR sets up the backend manager. The logic to transition to one provider independent manager was added, by modifying the already existing `JupyterDrivesManager` class.  To achieve this, the architecture for the `entrypoints` was modified, as we only need to register one manager for all providers now.  

The listing drives functionality was updated to have the provider independent set up. 

The `mount_drive` functionality was added in the backend and its corresponding frontend request. It is meant to be used when first accessing a drive, in order to create the content manager/store for it in the backend. Consequently, the `mounted` property was added to `IDriveInfo` interface to signal whether a content manager was already set up in the backend for the specific drive.

Also, switched to the use of the `obstore` package for the backend.

### CODE REVIEW

File: jupyter_drives/base.py [hunk 1]
```diff
@@ -7,9 +7,14 @@
 # Supported third-party services
 MANAGERS = {}
 
+# Moved to the arhitecture of having one provider independent manager.
```
Comment (afshin) [inline jupyter_drives/base.py:10]:
```suggestion
# Moved to the architecture of having one provider independent manager.
```

File: jupyter_drives/manager.py [hunk 2]
```diff
@@ -50,19 +52,59 @@ def per_page_argument(self) -> Optional[Tuple[str, int]]:
...
+                drives = [S3Drive(self._config.access_key_id, self._config.secret_access_key)]
+
+            elif self._config.provider == 'gcs':
+                GCSDrive = get_driver(Provider.GOOGLE_STORAGE)
+                drives = [GCSDrive(self._config.access_key_id, self._config.secret_access_key)] # verfiy credentials needed
+            
+            else: 
+               response = {
+                    "message": "Listing drives not supported for given provider.",
+                    "code": 501
```
Comment (afshin) [inline jupyter_drives/manager.py:76]: Can you get this from `httpx.codes` for readability?

File: jupyter_drives/manager.py [hunk 3]
```diff
@@ -50,19 +52,59 @@ def per_page_argument(self) -> Optional[Tuple[str, int]]:
...
+                        "name": result.name,
+                        "region": self._config.region_name if self._config.region_name is not None else "eu-north-1",
+                        "creation_date": result.extra["creation_date"],
+                        "mounted": "true" if result.name not in self._content_managers else "false",
+                        "provider": self._config.provider
+                    }
+                )
+            response = {
+                "data": data,
+                "code": 200
```
Comment (afshin) [inline jupyter_drives/manager.py:96]: Can you get this from `httpx.codes` for readability?

**Review by afshin — APPROVED** [review]
Thank you!

File: jupyter_drives/manager.py (hunk 2 above)
Comment (DenisaCG) [reply jupyter_drives/manager.py]: Latest commit switches fully to `httpx.codes`.

File: jupyter_drives/manager.py (hunk 3 above)
Comment (DenisaCG) [reply jupyter_drives/manager.py]: Latest commit switches fully to `httpx.codes`.

**Pull request merged**